shared/actions/wallets.tsx

```diff
@@ -1643,22 +1652,25 @@ function* loadStaticConfig(state: TypedState, action: ConfigGen.DaemonHandshakeP
       version: action.payload.version,
     })
   )
-  const loadAction = yield RPCStellarTypes.localGetStaticConfigLocalRpcPromise().then(res =>
-    WalletsGen.createStaticConfigLoaded({
-      staticConfig: I.Record(res)(),
-    })
-  )
 
-  if (loadAction) {
-    yield Saga.put(loadAction)
+  try {
+    const res: Saga.RPCPromiseType<
+      typeof RPCStellarTypes.localGetStaticConfigLocalRpcPromise
+    > = yield RPCStellarTypes.localGetStaticConfigLocalRpcPromise()
+    yield Saga.put(
+      WalletsGen.createStaticConfigLoaded({
+        staticConfig: I.Record(res)(),
+      })
+    )
+  } finally {
+    yield Saga.put(
```

Comment:
Did you mean to move this into `finally`?

Reply:
yes, these calls where we call handshakeWait true/false should always be balanced. else if it throws you'll be stuck forever